docs/content/develop/resource.md

```diff
@@ -0,0 +1,455 @@
+1. For each API field, copy the following template into the resource's `properties` attribute. Be sure to indent appropriately.
+
+```yaml
+# Supported types: String, Integer, Boolean, Double, Enum,
+# ResourceRef (link to a GCP resource), KeyValuePairs (string -> string map),
+# Array, and NestedObject
+- !ruby/object:Api::Type::String
+  name: 'API_FIELD_NAME'
+  description: |
+    MULTILINE_FIELD_DESCRIPTION
+  # Marks the field (and any subfields) as beta-only. Ensure a beta version block
+  # is present in provider.yaml. Do not use if an ancestor field (or the overall
+  # resource) is already marked as beta-only.
+  # min_version: beta
+
+  # If true, the field (and any subfields) will be considered immutable - that
+  # is, only settable on create. If unset or false, the field will still be
+  # considered immutable if any ancestor field (or the overall resource) is
+  # immutable, unless `update_url` is set.
+  # immutable: true
+
+  # If set, changes to the field's value will trigger a separate call to a
+  # specific API method for updating the field's value. The field will not be
+  # considered immutable even if an ancestor field (or the overall resource) is
+  # immutable.
+  # Terraform field names enclosed in double curly braces will be replaced
+  # with the field values from the resource.
+  # update_url: 'projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}/setFieldName'
+
+  # Overrides the verb used to update this specific field. Allowed values:
+  # :POST, :PUT, :PATCH. Default: Resource's update_verb (which defaults to :PUT
+  # if unset).
+  # update_verb: :POST
+
+  # If true, the field is required. If unset or false, the field is optional.
+  # required: true
+
+  # If true, the field is output-only - that is, it cannot be configured by the
+  # user. If unset or false, the field is configurable.
+  # output: true
+
+  # If true, the provider will set the field's value in the resource state based
+  # only on the user's configuration. If false or unset, the provider will set
+  # the field's value in the resource state based on the API response. This
+  # should only be used if the field cannot be read from GCP due to either API or
+  # provider constraints.
+  # ignore_read: true
+
+  # Sets a client-side default value for the field. This should be used if the
+  # API has a default value that applies in all cases and is stable. Removing
+  # or changing a default value is a breaking change.
+  # default_value: DEFAULT_VALUE
+
+  # If true, and the user has not set a value for this field in their
+  # configuration, the provider will accept any value returned from the API as
+  # the value for the field. If false, the provider will show a diff if the API
+  # returns a value and none is set in the user's config. This is useful for
+  # complex or frequently-changed API-side defaults, but provides less useful
+  # information at plan time than default_value.
+  # default_from_api: true
```

Comment:
If this value is set on a field, updating to the zero value for the type is impossible.

Reply:
clarified in 1a6b282b5

Reply:
"the provider will show a diff if the API returns a non-empty value for the field" is true if no default is set, but the diff will happen if anything but the default is returned if one is set.

Reply:
Good catch, thanks! Cleaned this up in e39e816fc